component/core/src/main/java/org/exoplatform/social/core/manager/ActivityManagerImpl.java

```
@@ -867,10 +877,67 @@ public boolean isActivityEditable(ExoSocialActivity activity, org.exoplatform.se
     return false;
   }
 
+  @Override
+  public String[] getActivityFilesIds(ExoSocialActivity activity) {
+    List<String> values = new ArrayList<>();
+    if (activity != null) {
+      String activityId = activity.getId();
+      String[] ids = getParameterValues(activity.getTemplateParams(), ID);
+      String[] storages = getParameterValues(activity.getTemplateParams(), STORAGE);
+      if (ids != null && ids.length > 0) {
+        for (int i = 0; i < ids.length; i++) {
+          if (storages != null && storages.length > 0) {
+            if (storages.length > i && storages[i].equals(FILE)) {
+              values.add(ids[i]);
+            }
+          }
+        }
+      }
+    }
+    return values.stream().toArray(String[]::new);
+  }
+
+  @Override
+  public ActivityFile getActivityFileById(long fileId) throws Exception {
+    FileItem file = null;
+    try {
+      file = fileService.getFile(fileId);
+    } catch (FileStorageException e) {
+      return null;
```

We should not swallow the exception, please add a log error.

---

done